packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -464,13 +465,68 @@ class TechnologyController {
 		return technology;
 	}
 
-	async finalizeRegistration({ params }) {
+	async finalizeRegistration({ params, request, auth }) {
 		const technology = await Technology.findOrFail(params.id);
+		const { comment } = request.all();
+		if (comment) {
+			const user = await auth.getUser();
+			const technologyComment = await TechnologyComment.create({ comment });
+			await Promise.all([
+				technologyComment.technology().associate(technology),
+				technologyComment.user().associate(user),
+			]);
+			await technology.load('comments');
+		}
 		technology.status = technologyStatuses.PENDING;
 		await technology.save();
 		Bull.add(Job.key, technology);
 		return technology;
 	}
+
+	async sendEmailToReviewer(technology, comment = null, antl) {
+		const reviewer = await technology.getReviewer();
+		const userReviewer = await reviewer.user().first();
+		const { from } = Config.get('mail');
+		try {
+			await Mail.send(
+				'emails.changes-made',
+				{ userReviewer, technology, comment },
+				(message) => {
+					message.subject(antl('message.reviewer.changesMade'));
+					message.from(from);
+					message.to(userReviewer.email);
+				},
+			);
+		} catch (exception) {
+			// eslint-disable-next-line no-console
+			console.error(exception);
+		}
+	}
+
+	async sendToRevision({ params, request, auth }) {
+		const technology = await Technology.findOrFail(params.id);
+		const { comment } = request.all();
+		if (comment) {
```

Comment:
Fiquei com um pé atrás em relação a esse if. Se o comentário é obrigatório (e not null no banco), pq deveria haver essa verificação? 🤔

Reply:
@mateus4k Vou explicar mais detalhadamente como é o fluxo aqui. 
1. O pesquisador vai enviar para revisão a tecnologia após receber um pedido de alterações e **não deseja enviar um comentário**.Nesse primeiro cenário ele não vai executar o código do if (comment), ou seja, ele não vai criar nenhuma entrada na tabela `technology_comments`, ele irá apenas mudar o status da tecnologia para CHANGES_MADE e enviar um email para o curador informando que ele fez as mudanças e deseja nova revisão.
2. O pesquisador vai enviar para revisão a tecnologia após receber um pedido de alterações e **deseja enviar um comentário**.Nesse segundo cenário ele vai executar o código do if (comment), ou seja, ele vai criar uma entrada na tabela `technology_comments` e nesse caso o comment é obrigatório, pois a tabela só possui um campo editável que é o próprio comment (não faz sentido criar um comment se o comentário é nulo, por isso ele é notNullable). Após isso ele vai executar o mesmo código do primeiro cenário.

Reply:
Obrigado pelo esclarecimento! 💯